packages/airnode-node/src/api/index.ts

```diff
@@ -190,14 +219,21 @@ async function processSuccessfulApiCall(
       reservedParameters as adapter.ReservedParameters
     );
 
-    if (aggregatedApiCall.type === 'testing-gateway') {
-      // NOTE: Testing gateway will use only the value and ignore the signature so there is no need
-      // to compute it, since it is performance heavy operation.
-      return [[], { success: true, value: JSON.stringify(response), signature: 'not-yet-supported' }];
+    switch (aggregatedApiCall.type) {
+      case 'http-gateway':
+        // NOTE: Testing gateway will use only the value and ignore the signature so there is no need
+        // to compute it, since it is performance heavy operation.
+        return [[], { success: true, value: JSON.stringify(response), signature: 'not-yet-supported' }];
+      case 'regular': {
+        const signature = await signWithRequestId(aggregatedApiCall.id, response.encodedValue, config);
+        return [[], { success: true, value: response.encodedValue, signature }];
+      }
+      case 'http-signed-data-gateway': {
+        const timestamp = Math.floor(Date.now() / 1000).toString();
+        const signature = await signWithTemplateId(parameters._templateId, timestamp, response.encodedValue, config);
+        return [[], { success: true, value: JSON.stringify({ timestamp, value: response.encodedValue }), signature }];
```

Comment:
The timestamp will be used while pushing the data to the chain so we need to return it to the caller

Reply:
The timestamp is there, you can see the resulting object example here https://github.com/api3dao/airnode/pull/870#pullrequestreview-893537683.
But as @Siegrift says, the format of different response types will be united in #871. I just don't want to add it to this PR as it's big enough as it is.

Reply:
Yeah sorry, I didn't see it for some reason